perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2201,22 +2201,41 @@ public void sendPasswordResetLinkEmail(PerunSession sess, Member member, String
 		for (Attribute a : logins) {
 			if (a.getFriendlyNameParameter().equals(namespace)) found = true;
 		}
-		if (!found) throw new InternalErrorException(user.toString()+" doesn't have login in namespace: "+namespace);
+		if (!found)
+			throw new InternalErrorException(user.toString() + " doesn't have login in namespace: " + namespace);
 
 		String email = "";
 		try {
-			Attribute a = perunBl.getAttributesManagerBl().getAttribute(sess, user, AttributesManager.NS_USER_ATTR_DEF+":preferredMail");
+			Attribute a = perunBl.getAttributesManagerBl().getAttribute(sess, user, mailAttributeUrn);
 			if (a != null && a.getValue() != null) {
-				email = (String)a.getValue();
+				email = (String) a.getValue();
 			}
 		} catch (WrongAttributeAssignmentException ex) {
 			throw new InternalErrorException(ex);
 		} catch (AttributeNotExistsException ex) {
 			throw new InternalErrorException(ex);
 		}
 
+		String subject;
+		try {
+			Attribute messageTemplateAttribute = perunBl.getAttributesManager().getAttribute(sess, language,
+					"urn:perun:entityless:attribute-def:def:nonAuthzPwdResetMailSubject:" + namespace);
+			subject = messageTemplateAttribute.getValue().toString();
+		} catch (PrivilegeException | AttributeNotExistsException | WrongAttributeAssignmentException ex) {
+			throw new InternalErrorException(ex);
+		}
+
+		String message;
+		try {
+			Attribute subjectTemplateAttribute = perunBl.getAttributesManager().getAttribute(sess, language,
+					"urn:perun:entityless:attribute-def:def:nonAuthzPwdResetMailTemplate:" + namespace);
+			message = subjectTemplateAttribute.getValue().toString();
```

Comment:
Since attribute value can be null its not safe to call ``.toString()`` on it. You should use ``message = (String) subjectTemplateAttribute.getValue()`` when assigning. I know IDE then complains on unsafe cast, but its safe since we know which attribute we use here.

Reply:
If you want, you can then suppress the warning and write comment why this check is ok so when anyone else sees it, he knows its fine.